### ISSUE TITLE
Custom fields appear do not appear on the tree detail page by default

### ISSUE THREAD

**Issue by dboyer** [OpenTreeMap/otm-core#2871]
When adding a new custom field, the user currently needs to:

1. Add the field on the custom fields page (may include setting permissions)
2. Set access permissions via the Roles page if not completed on the Custom Fields page
3. Go to the Data Fields page and set the fields to display on the detail page 

I would like to have the fields display by default. The user can go to the Data Fields page to set them to not display if desired.

**Comment by dboyer** [comment]
Now I can't reproduce this and they are appearing by default.

**Comment by RobinIsTheBird** [comment]
@dboyer, I'll try to reproduce it.

**Comment by RobinIsTheBird** [comment]
I could not reproduce this.

My local database is brand new, so there wouldn't be any caching on the server.

@dboyer and @jwalgran, what do you suggest we do with this issue? Should we close it?

**Comment by jwalgran** [comment]
I was able to reproduce this on staging by following these steps

- Create a new instance centered on Los Angeles
- Add a decimal field to the Tree model
- Update the permissions on the field to make it writable by everyone

The field was not checked on the `/{instance}/management/field-configuration/` page and did not appear on the tree detail page.

**Comment by RobinIsTheBird** [comment]
@dboyer, here is what I found, and what the PR I've made does.

I found code to enable udfs by default, but it was incorrect. It only enabled a newly created udf by default if one other udf already in existence had already been manually enabled.

I found no evidence of code to set permissions to make the new udf visible to _any_ role. I did find code to restrict it to be invisible to all roles, with a comment emphasizing that intent.

My PR corrects the udf enabling code to work the first time a udf is created, and assigns udf permissions per role to be the same as the role's default permission level. So udfs will be immediately visible for all our default roles - administrator, editor, and public.

There is presently no way (afaik) for an administrator to change the default permission level of a role. I think we now have enough code dependent on that default permission level (visible udfs, add/delete, external link, modeling...) that we should consider creating an issue to make it settable.

**Comment by maurizi** [comment]
@dboyer @RobinIsTheBird 

The intended behavior for the Roles page was that newly added fields are by default Invisible.

When we added the data fields page, we decided to make things be enabled there by default, so that map owners would not have to go to _2_ places to enable a newly added field.

If we want to change the behavior of what happens on the Roles page, I think that should be done as a separate issue that does not block the release, as there is more to be done than just changing the code to use the default permissions (need to change help text, consider removing yellow notification bubbles, etc.)

**Comment by RobinIsTheBird** [comment]
@maurizi, where is the help text that would change?

I think the yellow notification bubbles still apply, and their text is still relevant.

**Comment by dboyer** [comment]
I agree that the release blocking issue is to make newly added custom fields appear on the tree detail page by default. Thanks for fixing that issue, Robin.

Changing the default behavior of Roles is a separate issue and one that I would not want to tackle without some additional discussion. Our standard has previously been to make custom fields invisible to all roles, unless an administrator specifically changes that, in order to prevent users with a lower access level from making changes they should not make.

**Comment by maurizi** [comment]
Oh huh nevermind on the help text - "You may want to change the default permissions" work whether they are invisible or not.

**Comment by RobinIsTheBird** [comment]
@dboyer, could you please clarify?

On the one hand, you say that the release blocking issue is to make newly added custom fields appear on the tree detail page by default.

On the other hand, you mention that our standard has previously been to make custom fields invisible to all roles by default in the same paragraph as saying that changing the default behavior of Roles is a separate issue requiring more discussion.

So it sounds like you want them to show but not have visibility permission right now, which is a contradiction. Could you please clarify what you want to happen for 2.13.0?

**Comment by dboyer** [comment]
I would like them to appear on the tree detail page by default but respect any visibility permissions that exist. The main goal is to skip Step 3 in the the three step process described in the initial issue I logged and repeated below.

Currently, admins have to:

1. Add the field on the custom fields page (may include setting permissions)
2. Set access permissions via the Roles page if not completed on the Custom Fields page
3. Go to the Data Fields page and set the fields to display on the detail page

I would like to skip Step 3 and have the fields display on the detail page by default. If the admin wants them to not display by default, they would then go to the Data Fields page and change that option. We should still respect visibility permissions. If the admin wants only admins and editors to see the field, it would be invisible to public users.

**Comment by RobinIsTheBird** [comment]
tl;dr

@dboyer, the question is about the defaults.

Both the custom fields page and the Roles page let you change the udf permission per role, to FULL WRITE ACCESS, READ ONLY, or INVISIBLE.

When you are on the custom fields page, and you add a new custom field, and click Finish on the dialog, you then see the following in production or staging as they now stand:

![new custom field](https://cloud.githubusercontent.com/assets/1473263/20681061/d778d32e-b56f-11e6-964a-a6d79333b9d7.png)

In production or staging, if you then navigate to the Roles page, you also see "**Invisible**" straight across for the new custom field.

Based on my understanding of the issue description, I changed it so that on the Custom Fields page, where you currently see Invisible three times, you would see **Full Write Access** for _Administrator_ and _Editor_, and **Read Only** for _Public_, which are the `default_permission_level` settings for each of those roles. If any custom roles have been added, they would appear as **Invisible** because the `default_permission_level` for custom roles is **Invisible** and there is no way to change the default permission level in the UI.

Consistenly, in my PR, navigating to the Roles page would show you those permissions for those roles for the new custom field. The Roles page would still give you the same yellow warning, saying, "New fields have been added and are highlighted below. You may want to change the default permissions.", and the new custom field row would still be highlighted in gold, same as before.

So, after saying all that, when you click **Finish** on the New Custom Field dialog, and are then shown the roles and their permissions for the new field, what set of permissions should it show?
